File: Data Structures & Algorithms/reverse-integer/submission-1.java
class Solution {
    public int reverse(int x) {
        int ret = 0;
        while(x != 0){
            int prevRet = ret;
            ret *= 10;
            int digit = x%10;
            ret += digit;
            x/=10;
            if((ret - digit) / 10 != prevRet) return 0;
        }
        return ret;
    }
}

File: Data Structures & Algorithms/reconstruct-flight-path/submission-1.py
class Solution:
    def findItinerary(self, tickets: List[List[str]]) -> List[str]:
        # build adj_list

        graph = {}

        for src, dst in tickets:
            if src not in graph:
                graph[src] = []
            graph[src].append(dst)

        for k,v in graph.items():
            v.sort(reverse=True)
        
        stack = ['JFK']
        ret = []

        while stack:
            nxt = stack[-1]
            
            if nxt in graph and graph[nxt]:
                nn = graph[nxt].pop()
                stack.append(nn)
            else:
                ret.append(stack.pop())
        return ret[::-1]
            
        